scripts/configure.py

```diff
@@ -0,0 +1,136 @@
+    try:
+        from pip._internal import main
+    except Exception:
+        from pip import main
+    # install requirement or MAR-INF/requirements.txt
+    if not requirement:
+        main(['install', package_name, '-t', target])
+    else:
+        main(['install', '-r', package_name, '-t', target])
+
+
+def download_url(url, target, retries=2, base_retry_interval=0.01):
+    """Download url to a file.
+    Parameters
+    ----------
+    url: string
+        url to the file to download
+    target: string
+        the target local path of the downloaded file
+    retries: int
+        the max number of retries allowed for urlretrieve
+    """
+    assert retries >= 1, "Number of retries should be at least 1"
+
+    retry = 0
+    while retry < retries:
+        try:
+            urllib.urlretrieve(url, target)
+            return
+        except Exception as e:
+            retry += 1
+            if retry < retries: 
+                time.sleep(2 ** retries * base_retry_interval)
+            else:
+                raise e
+
+
+def check_existence(filename, path):
+    """Check the existence of filename in path
+
+    Parameters
+    ----------
+    filename: string
+        name of file or dir to verify existence
+    path: string
+        path to verify the existence
+
+    Returns
+    -------
+    bool: Returns True if exist, False otherwise
+    
+    Examples
+    ----------
+    >>> check_existence('mxnet/', lambda_function_path)
+    >>> check_existence('MAR-INF/requirements.txt', lambda_function_path)
+    """
+    return os.path.exists(os.path.join(path, filename))
+
+
+def configure(model_archive, lambda_function_path, model_bucket=None):
+    """Configure Lambda Function to consume Model Archive
+    
+    Parameters
+    ----------
+    model_archive: string
+        url or path to the local model archive file
+    model_bucket: string
+        optional S3 bucket name to upload local model archive file
+    """
+    # check weither it is url
+    is_url = "://" in model_archive
+    url_mar = None
+    if is_url==False and model_bucket==None:
+        raise Exception("Please specify S3 bucket for model storage")
+    # create a tmp path
+    dirpath = tempfile.mkdtemp()
+    # download if is url
+    if is_url:
+        download_url(model_archive, os.path.join(dirpath, "tmp.mar"))
+        url_mar = model_archive
+        model_archive = os.path.join(dirpath, "tmp.mar")
+    # unzip
+    status = subprocess.call(" ".join(["unzip", model_archive, "-d", dirpath]), shell=True)
+    if status != 0:
+        raise Exception("Failed to unzip model archive. Interrupt.")
+    # install requirements
+    if check_existence('MAR-INF/requirements.txt', dirpath):
+        do_install(os.path.join(dirpath, 'MAR-INF/requirements.txt'), requirement=True, target=lambda_function_path)
+    if check_existence('mxnet/', dirpath) == False and check_existence('mxnet/', lambda_function_path) == False:
```

Comment:
can you please add comments on why this is needed(numpy > 1.13 does not work with certain versions of mxnet)

Reply:
forgot to push? I don't see the change.

Reply:
scratch that, found it below